### PR TITLE
Created Issue and PR Templates

### LINKED ISSUE #69: Add issue and PR tempaltes
2 issue templates 
1.  Bugs
2. Feature Requests

And a template with a basic checklist for PRs is needed

### PR DESCRIPTION
**Pull request by dumbcoder7**
fixes #69

### CODE REVIEW

**Comment by ADI10HERO** [comment]
We are yet to decide on a template and surely don't want all of them assigned to me.
We'll get back on this real soon 👍 


Anyway, thank you for the thoughtful contribution 😄

**Comment by dumbcoder7** [comment]
Oh okay then, I'll remove your name. Should I close the PR or let it stay?

**Comment by ADI10HERO** [comment]
> Oh okay then, I'll remove your name. Should I close the PR or let it stay?

Let the PR stay, I have a template in mind, would love to go over it together on [Slack](autodl.slack.com)

**Comment by dumbcoder7** [comment]
Okay cool

**Comment by dumbcoder7** [comment]
@ADI10HERO Review and suggest.

File: .github/ISSUE_TEMPLATES/Feature Request.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+about: Suggest an idea for this project
+title: "[FEATURE]"
+labels: ''
+assignees: ''
+
+---
+
+## Feature Request
+
+- [ ] Related to an existing [Issue](../../../issues)
```
Comment (ADI10HERO) [inline .github/ISSUE_TEMPLATES/Feature Request.md:12]:
```suggestion
- [ ] Related to an existing [Issue](../issues) 
```

Comment (dumbcoder7) [reply .github/ISSUE_TEMPLATES/Feature Request.md]: I think both paths were wrong. Added the correct one.

File: .github/ISSUE_TEMPLATES/Feature Request.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+- [ ] Related to an existing [Issue](../../../../issues) 
+- [ ] A new implementation (Improvement, Extension) 
+
+### Is your feature request related to a problem?
+
+A clear and concise description of what the problem is. For e.g. I have an issue when [...]
+
+### If your feature will improve `Auto-DL`
+
+A clear and concise description of how it will help `Auto-DL`. Please prefer references, if possible [...]
```
Comment (ADI10HERO) [inline .github/ISSUE_TEMPLATES/Feature Request.md:21]:
```suggestion
### Feature Description
<!-- A clear and concise description of the feature you're requesting -->
```

File: .github/ISSUE_TEMPLATES/Feature Request.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+### Description of the possible solution
+
+A clear and concise description of what you want to happen. Add any considered drawbacks.
+
+### An alternative solution to the problem can look like
+
+A clear and concise description of any alternative solutions or features you've considered.
```
Comment (ADI10HERO) [inline .github/ISSUE_TEMPLATES/Feature Request.md:29]:
```suggestion
### Possible Solution
<!-- We would love to know if you can suggest a solution to the feature you're requesting, if nothing else it'll help us understand your expectations -->
```

File: .github/ISSUE_TEMPLATES/Feature Request.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+**Teachability, Documentation, Adoption, Migration Strategy**
+If you can, explain how users will be able to use this and possibly write out a version of the docs.
+Can you add a screenshot or design?
```
Comment (ADI10HERO) [inline .github/ISSUE_TEMPLATES/Feature Request.md:33]:
```suggestion
### Additional context (optional)
<!-- If you can, explain how users will be able to use this and possibly write out a version of the docs or add a screenshot or design for the same -->
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+## Pull Request
+
```
Comment (ADI10HERO) [inline .github/PULL_REQUEST_TEMPLATE.md:2]:
```suggestion

fixes #  
<!-- Please add the issue number you're trying to fix with this PR, if none then please create an issue first -->
### Description
<!-- A line or two explaining what this patch does -->
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
+## Pull Request
```
Comment (ADI10HERO) [inline .github/PULL_REQUEST_TEMPLATE.md:1]:
```suggestion

---
name: PR template
about: Submit a patch to fix issues and improve Auto-DL
title: "[Fixes #] "
labels: ''
assignees: ''
---

## Pull Request
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+### All Submissions:
+
+- [ ] Have you followed the guidelines mentioned in our [Contributing document](/CONTRIBUTING.md)?
+- [ ] Have you ensured that there aren't other open [Pull Requests](../../../pulls) for the same update/change?
+- [ ] Is your [Pull Request](../../../pulls) linked to an [Issue](../../../issues)? If yes, please mention the Issue Number.
```
Comment (ADI10HERO) [inline .github/PULL_REQUEST_TEMPLATE.md:7]:
```suggestion
### Checklist:
- [ ] Followed the guidelines mentioned in our [Contributing document](/CONTRIBUTING.md)
- [ ] Ensured that there aren't other open [Pull Requests](../../../pulls) for the same update/change
- [ ] This PR is linked to an [Issue](../../../issues) and the issue number has been mentioned in the title and body of the PR 
- [ ] The code is clean and docstrings have been added or updated as required
- [ ] Code is linted/formatted locally prior to submission (using `black`)
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+- [ ] Have you described your new feature in the `docs`?
+
+### Changes to Core Features:
+
+- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
+- [ ] Have you written new `tests` for your core changes, as applicable?
+- [ ] Have you successfully ran `tests` with your changes locally?
+- [ ] Have you kept the `coverage-rate` constant?
+- [ ] Have you managed to increase the `coverage-rate`?
+- [ ] Are the `docs` up to date?
```
Comment (ADI10HERO) [inline .github/PULL_REQUEST_TEMPLATE.md:23]:
```suggestion
```

**Review by ADI10HERO — CHANGES_REQUESTED** [review]
Hey @dumbcoder7 

I have proposed the changes, my goal was to make the template as easy as possible and removed a lot of checks, especially from the PR template.

Please go through them, feel free to continue the discussion either here or on slack.

Thanks,
Adi

File: .github/ISSUE_TEMPLATES/Bug Report.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
+---
+name: Bug report
+about: Create a report to help us improve
+title: "[BUG]"
```
[the remark this answers is not in the document]
Comment (dumbcoder7) [reply .github/ISSUE_TEMPLATES/Bug Report.md:4]: I think you suggested this one.

File: .github/ISSUE_TEMPLATES/Bug Report.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+labels: ''
```
[the remark this answers is not in the document]
Comment (dumbcoder7) [reply .github/ISSUE_TEMPLATES/Bug Report.md:5]: Oh okay. Either in title or label. Label was my first choice too.

File: .github/ISSUE_TEMPLATES/Bug Report.md [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+### ⚙️ Environment
+
+- `Python` version(s):
+    - *good*: [e.g. 3.8]
+    - **better**: [3.8.6 - Clang 12.0.0 (clang-1200.0.32.27)]
+- `OS`: [e.g. Ubuntu 18.04, OSX 10.13.4, Windows 10]
+- `IDE`: [Vim, VS-Code, Pycharme]
+- `Packages`: [ `Tensorflow==2.1.2 - latest`]
```
[the remark this answers is not in the document]
Comment (dumbcoder7) [reply .github/ISSUE_TEMPLATES/Bug Report.md:30]: Yeah I just basically pasted it as it is, as I was not aware of what's needed and what's not.

File: .github/ISSUE_TEMPLATES/Bug Report.md [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+
+
+### Possible Solution  (optional)
```
[the remark this answers is not in the document]
Comment (dumbcoder7) [reply .github/ISSUE_TEMPLATES/Bug Report.md:31]: According to me, not everyone who finds a bug will be able to provide an effective solution. What do you think?

[the remark this answers is not in the document]
Comment (ADI10HERO) [reply .github/ISSUE_TEMPLATES/Bug Report.md:31]: I agree to this, but having that `(optional)` in the title is weird, isn't its?

Can we write ...
`<!-- Optional section (feel free to remove it) -->` 
something like this in the comments

File: .github/ISSUE_TEMPLATES/Feature Request.md [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+name: Feature Request
+about: Suggest an idea for this project
+title: "[FEATURE]"
+labels: ''
+assignees: ''
+
+---
+
+## Feature Request
+
```
[the remark this answers is not in the document]
Comment (dumbcoder7) [reply .github/ISSUE_TEMPLATES/Feature Request.md:11]: I'd rather convert this to a question. The instruction might not go well with it.

`Is this related to an existing [Issue] or a new implementation? `

[the remark this answers is not in the document]
Comment (ADI10HERO) [reply .github/ISSUE_TEMPLATES/Feature Request.md:11]: That works as well, just that people should know that putting an `x` between the brackets will add a tick mark to the box there :D

File: .github/ISSUE_TEMPLATES/Bug Report.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (dumbcoder7) [reply .github/ISSUE_TEMPLATES/Bug Report.md:31]: Yep okay. Will change.

File: .github/ISSUE_TEMPLATES/Feature Request.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (dumbcoder7) [reply .github/ISSUE_TEMPLATES/Feature Request.md:11]: Nah I am saying remove the checkbox cuz it's a binary question here. It doesn't really belong. Only the question remains.

File: .github/ISSUE_TEMPLATES/Bug Report.md [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+## Bug Report
+
+
+
+### Current Behavior
+A clear and concise description of the behavior.
+
+If possible please provide a minimal code sample or steps to replicate the issue 
+
+**Expected behavior/code**
```
Comment (ADI10HERO) [inline .github/ISSUE_TEMPLATES/Bug Report.md:19]:
```suggestion
### Expected behavior/code
```

File: .github/ISSUE_TEMPLATES/Bug Report.md [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+---
+
+## Bug Report
+
+
+
+### Current Behavior
+A clear and concise description of the behavior.
+
+If possible please provide a minimal code sample or steps to replicate the issue
```
Comment (ADI10HERO) [inline .github/ISSUE_TEMPLATES/Bug Report.md:17]:
```suggestion
<!-- A clear and concise description of the behavior.  If possible please provide a minimal code sample or steps to replicate the issue -->
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 16]
```diff
@@ -0,0 +1,21 @@
+
+---
+name: PR template
+about: Submit a patch to fix issues and improve Auto-DL
+title: "[Fixes #] "
+labels: ''
+assignees: ''
+---
```
Comment (ADI10HERO) [inline .github/PULL_REQUEST_TEMPLATE.md:8]:
```suggestion
---
name: PR template
about: Submit a patch to fix issues and improve Auto-DL
title: "[Fixes #] "
labels: ''
assignees: ''

---
```

File: .github/ISSUE_TEMPLATES/Feature Request.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (dumbcoder7) [reply .github/ISSUE_TEMPLATES/Feature Request.md:11]: Changes incorporated

**Review by ADI10HERO — APPROVED** [review]

**Pull request merged**